spring-cloud-dataflow-registry/src/main/java/org/springframework/cloud/dataflow/registry/service/DefaultAppRegistryService.java

```diff
@@ -141,10 +141,19 @@ public AppRegistration save(String name, ApplicationType type, String version, U
 
 	@Override
 	public AppRegistration save(AppRegistration app) {
+		AppRegistration appRegistration = this.appRegistrationRepository.findAppRegistrationByNameAndTypeAndVersion(
+				app.getName(), app.getType(), app.getVersion());
 		if (getDefaultApp(app.getName(), app.getType()) == null) {
```

Comment:
I guess setting a default on first registration can go inside the `else` block as it affects only new application registrations